src/LanguageServer/Impl/Diagnostics/DiagnosticsService.cs

```diff
@@ -36,11 +36,18 @@ private sealed class DocumentDiagnostics {
             public bool Changed { get; set; }
 
             public void Clear(DiagnosticSource source) {
-                Changed = Entries.Count > 0;
+                if (!_entries.TryGetValue(source, out _)) {
+                    return;
+                }
+
+                Changed = true;
                 _entries[source] = Array.Empty<DiagnosticsEntry>();
             }
 
-            public void ClearAll() => _entries.Clear();
+            public void ClearAll() {
+                Changed = true;
```

Comment:
this one seems a bug. it cleared all code but never but it as Changed. so published diagnostic probably won't get ever cleaned up?